### PR TITLE
[Hub Generated] Review request for Microsoft.Authorization to add version preview/2020-04-01-preview

### PR DESCRIPTION
**Pull request by dagoroz**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/dagoroz/azure-rest-api-specs/dev-authorization-Microsoft.Authorization-2020-04-01-preview...master/).

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by dagoroz** [comment]
Hi @yungezz, I would like your opinion on this matter.
I believe this breaking change tag is a false positive because the error codes I added have always been part of our service it's just that they were never documented. (same for the new objects and properties)

Please let me know if this situation can indeed be categorized as a false positive because we have a backlog of swaggers we are planning to catch up to which will very likely trigger this exact situation over and over

**Comment by yungezz** [comment]
discussing in email.

**Review by JeffreyRichter — APPROVED** [review]

**Comment by yungezz** [comment]
hi @ArcturusZhang could you pls check impact to GO SDK for `CI-Breakingchange-Go`? thanks

**Review by ArcturusZhang — APPROVED** [review]
The breaking changes in go SDK are expected

**Comment by dagoroz** [comment]
@yungezz friendly ping :)

**Comment by dagoroz** [comment]
please ignore closing and reopening

**Comment by yungezz** [comment]
hi @changlong-liu could you pls help to look at the SDK Python CI task failure or help to route to right owner? thanks.

**Comment by changlong-liu** [comment]
add Python owner @msyyc to have a look, thanks!

**Comment by dagoroz** [comment]
@msyyc can you take a look at this please?

**Comment by msyyc** [comment]
> @msyyc Yuchao Yan FTE can you take a look at this please?

the reason is that the existing tag name is changed in readme.md, while the name is not changed in readme.python.md.(picture below) 
And I advise add a new tag rather than change the name of exsting tag.
image
image

**Review by yungezz — APPROVED** [review]

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json [hunk 1]
```diff
@@ -135,6 +135,12 @@
             "schema": {
               "$ref": "#/definitions/RoleAssignmentListResult"
             }
+          },
+          "400": {
+            "description": "Bad Request - Indicates the request has missing information and can not be completed",
+            "schema": {
+              "$ref": "#/definitions/Error"
+            }
```
Comment (ArcturusZhang) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:143]: Hi @dagoroz @yungezz could any one explain this to me, why we have to add these "error response code" into this list?
In the generated SDKs, the status code listed here will **NOT** be considered as errors.
To indicate errors, please put the errors in the `default` response.

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json [hunk 2]
```diff
@@ -463,6 +502,12 @@
...
           }
```
Comment (ArcturusZhang) [inline specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:511]: Also the error response here is causing autorest generation failure - our convention is that the `responses` only list the successful status code that would return something from the service, not these error responses.

**Review by ArcturusZhang — COMMENTED** [review]
Hi @dagoroz there are multiple errors in this merged PR, please open another PR to fix this, thanks

File: specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json (hunk 1 above)
Comment (weidongxu-microsoft) [reply specification/authorization/resource-manager/Microsoft.Authorization/preview/2020-04-01-preview/authorization-RoleAssignmentsCalls.json:143]: Same opinion as @ArcturusZhang. If 400 is expected as error, it should be covered by `default`, not `400`.

This change breaks track2 Java SDK, as `RoleAssignmentListResult` and `Error` is 2 completely different type. Hence the method will not able to figure out which type will be returned in runtime.

**Pull request merged**